dbusmock/templates/systemd-base.py

```diff
@@ -0,0 +1,128 @@
+'''systemd mock template base
+'''
+
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the Free
+# Software Foundation; either version 3 of the License, or (at your option) any
+# later version.  See http://www.gnu.org/copyleft/lgpl.html for the full text
+# of the license.
+
+__author__ = 'Jonas Ådahl'
+__copyright__ = '(c) 2021 Red Hat'
+
+from gi.repository import GLib
+
+from dbusmock import MOCK_IFACE
+from dbusmock import mockobject
+
+BUS_PREFIX = 'org.freedesktop.systemd1'
+PATH_PREFIX = '/org/freedesktop/systemd1'
+
+
+BUS_NAME = BUS_PREFIX
+MAIN_OBJ = PATH_PREFIX
+MAIN_IFACE = BUS_PREFIX + '.Manager'
+
+
+UNIT_IFACE = BUS_PREFIX + '.Unit'
+
+
+def load_base(mock, _parameters):
+    mock.next_job_id = 1
+    mock.units = {}
+
+    mock.AddMethod(MAIN_IFACE,
+                   'StartUnit',
+                   'ss', 'o',
+                   'ret = self.StartUnit(self, *args)')
+    mock.AddMethod(MAIN_IFACE,
+                   'StartTransientUnit',
+                   'ssa(sv)a(sa(sv))', 'o',
+                   'ret = self.StartTransientUnit(self, *args)')
+    mock.AddMethod(MAIN_IFACE,
+                   'StopUnit',
+                   'ss', 'o',
+                   'ret = self.StopUnit(self, *args)')
+    mock.AddMethod(MAIN_IFACE,
+                   'GetUnit',
+                   's', 'o',
+                   'ret = self.GetUnit(self, *args)')
+    mock.AddProperties(MAIN_IFACE, {'Version': 'v246'})
+
+    mock.StartUnit = StartUnit
+    mock.StartTransientUnit = StartTransientUnit
+    mock.StopUnit = StopUnit
+    mock.GetUnit = GetUnit
+
+    mock.AddMethod(MOCK_IFACE, 'AddMockUnit', 's', '', 'self.AddMockUnit(self, *args)')
+
```

Comment:
Sooo... This can be reduced by just using `AddMethods` and annotations (that will help us also in handling the code better) via https://paste.ubuntu.com/p/gJS9td7HSR/

You add an even better implementation without having to repeat the signature though :)

Reply:
Well, we'll see what @martinpitt thinks about the new mockobject method :P

Reply:
I guess this can be marked as resolved, no idea how I can do it :stuck_out_tongue_closed_eyes: